### ISSUE TITLE
Emotion 10 intrinsics typescript types do not have event handler props

### ISSUE THREAD

**Issue by aaronjensen** [emotion-js/emotion#1088]
- `emotion` version: 10.0.4
- `react` version: 16.7.0-alpha.2

**Relevant code:**

```javascript
const Form = styled.form({})

const Foo = () => <Form onSubmit={e => { console.log(e) }} />
```

**What you did:**

Upgraded to Emotion 10

**What happened:**

Intrinsics created by `styled` do not appear to have event handler props like `onSubmit`/`onChange` so typescript complains about the implicit any of `e`.

```
error TS7006: Parameter 'event' implicitly has an 'any' type.
```

**Reproduction:**

Codesandbox doesn't import emotion types, so it's not helpful to repro. If there's a way to make that work, or if there's a starter project w/ Typescript + emotion 10 I'd gladly whip up a repro.

**Comment by Ailrun** [comment]
@aaronjensen Thank you for your issue! However, I cannot reproduce the issue. Could you please provide a reproduction?
I tried same code with
- typescript: 3.2.2
- react: 16.7.0-alpha.2
- @types/react: 16.7.13
- emotion: 10.0.4

**Comment by aaronjensen** [comment]
@Ailrun did you try with `noImplicitAny`? TypeScript by default would not complain about this. 

If so, let me know and I'll provide a repro. If you could push up your repro attempt somewhere that'd be helpful. Thanks!

**Comment by aaronjensen** [comment]
Sorry, one missing step:

```ts
import * as React from 'react'
import styled, { ObjectInterpolation } from '@emotion/styled'

export const flex = (): ObjectInterpolation<any> =>  ({})

const Search = styled.form(flex())

const Foo = () => <Search onSubmit={e => { console.log(e) }} />

export default Foo
```

Possibly related: https://github.com/emotion-js/emotion/issues/1067

**Comment by aaronjensen** [comment]
This should repro: 

https://github.com/aaronjensen/emotion-typescript-repro

```sh
yarn && yarn tsc
```

**Comment by Ailrun** [comment]
@aaronjensen OK, now I understood. It's because TS infer generic parameter as `any`...
```
export const flex = <T extends {}>(): ObjectInterpolation<T> =>  ({});

const Search = styled.form<{}>(flex());
```
will work but I don't think this is preferable... Do you have any opinion on this?

**Comment by aaronjensen** [comment]
@Ailrun yes, that works. It seems to have a slightly different type which causes autocomplete on the properties not to work as well (though type checking still does).

Compare:

```ts
export const flex = <T extends {}>():ObjectInterpolation<T> =>
  ({})

const Search = styled.form<{}>(flex())
const Foo = () => <Search onSubmit={e => { console.log(e) }} />

const Search2 = styled.form({})
const Foo2 = () => <Search2 onSubmit={e => { console.log(e) }} />
```

Is there a way to get a CSSProperties like type that isn't generic? TBH, I don't really understand the `MP` param. I just want to be able to specify that some of my utility functions return css properties (complete with `:hover`/other nesting capabilities).

**Comment by Ailrun** [comment]
@aaronjensen `MP` stands for MergedProps, and it's for representing props that is used in FunctionInterpolation.
Make `CSSObject` like following would work
```ts
type CSSPseudosForCSSObject = { [K in CSS.Pseudos]?: CSSObject };

export interface CSSOthersObjectForCSSObject {
  [propertiesName: string]: CSSObject;
}

export interface CSSObject
  extends CSSProperties,
    CSSPseudosForCSSObject,
    CSSOthersObjectForCSSObject {}
```

**Comment by aaronjensen** [comment]
It looks like that will work for me. I don't need it to accept functions because these are for functions that return styles.

By the way, `CSSPropertiesWithMutliValues` is misspelled `Mutli` vs `Multi`.

Looks like all I need now to upgrade to emotion 10 would be the above new type, `innerRef` support (I may be missing something here) and the babel jsx pragma eliding fix.

**Comment by Ailrun** [comment]
@aaronjensen Could you send a PR for this? I want to update one but recently I'm too busy... :cry:
